zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/AuthenticationInterceptor.java

```diff
@@ -55,6 +69,20 @@ public <ReqT, RespT> Listener<ReqT> interceptCall(
       // get user claims and set them in the context
       final JwtDecoder jwtDecoder = new JwtDecoder(token);
       final var claims = jwtDecoder.decode().getClaims();
+      final var username = getUsernameFromClaims(claims);
+      final var user = loadUserByUsername(username);
+      if (user.isEmpty()) {
+        LOGGER.debug(
+            "Denying call {} as the user {} does not exist",
+            methodDescriptor.getFullMethodName(),
+            username);
+        return deny(
+            call,
+            Status.UNAUTHENTICATED
+                .augmentDescription("Expected a valid user, but user does not exist")
+                .withCause(new IllegalArgumentException("User does not exist")));
+      }
+      claims.put(AUTHORIZED_USER_KEY, user.get().userKey());
```

Comment:
❌ As discussed on Slack. When we receive a bearer token we don't have to query ES. We must just validate the token and set all the claims on the token that goes to Zeebe.

Querying ES is only relevant for basic auth.